### PR TITLE
fix(LEMS-3482): add non-empty string validation in radio parser

### PR DESCRIPTION
**Pull request by anakaren-rojas**
## Summary:
- Creates defaulted function which handles empty strings as invalid
- Consumes new function in radio parser, sets `generateChoiceIds` function as fallback when existing ID is an empty string
- Adds tests that validate choice ids will not be empty strings and are properly updated when id is not present 

Issue: LEMS-3482

## Test plan:

https://github.com/user-attachments/assets/9c66ea80-0f3c-4978-af08-b632ef1f3282

### CODE REVIEW

File: packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts [hunk 1]
```diff
@@ -108,6 +108,160 @@ describe("parseRadioWidget", () => {
...
+        it("generates default ID when choice ID is missing", () => {
+            const widget = {
+                type: "radio",
+                version: {major: 3, minor: 0},
+                graded: true,
+                options: {
+                    choices: [
+                        {
+                            content: "Choice without ID",
+                            // id property is missing
```
Comment (mark-fitzgerald) [inline packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts]: This is testing for `undefined`, correct? Should there be an explicit test for `null`?

**Review by mark-fitzgerald — APPROVED** [review]
Quick fix. Thank you!

File: packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts (hunk 1 above)
Comment (anakaren-rojas) [reply packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts]: Added!

File: packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts [hunk 2]
```diff
@@ -108,6 +108,193 @@ describe("parseRadioWidget", () => {
...
+                                content: "Choice with empty ID",
+                                id: "radio-choice-0", // Should be generated
+                            },
+                        ],
+                    },
+                }),
+            );
+        });
+
+        it("generates default ID when choice ID is whitespace-only", () => {
```
Comment (benchristel) [inline packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts]: Do we actually need to treat whitespace-only IDs as invalid? It doesn't seem like a case we're likely to ever encounter.

File: packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts [hunk 3]
```diff
@@ -108,6 +108,193 @@ describe("parseRadioWidget", () => {
...
+                            {content: "First choice", id: "radio-choice-0"},
+                            {content: "Second choice", id: "custom-id"}, // preserved
+                            {content: "Third choice", id: "radio-choice-2"},
+                            {content: "Fourth choice", id: "radio-choice-3"},
+                        ],
+                    },
+                }),
+            );
+        });
+    });
```
Comment (benchristel) [inline packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts]: These tests cover all the cases, but they're a little verbose, and they're really testing `defaultedNonEmptyString`. I'd prefer to have comprehensive tests in `defaulted.test.ts` that call `defaultedNonEmptyString` directly, and then we'd only need to test a couple cases here (maybe the empty string case and the nonempty string case).

If there's published content that has blank choice IDs, we should also put an example of that in a regression test data file in `regression-tests/item-data`.

File: .changeset/funny-seahorses-reflect.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/perseus-core": patch
+---
+
+Radio fix that disallows choice ids that are empty strings
```
Comment (benchristel) [inline .changeset/funny-seahorses-reflect.md]: Can we say something like "defaults" here instead of "disallows"?

```suggestion
Radio fix that defaults choice ids that are empty strings
```

**Review by benchristel — APPROVED** [review]
I left a few suggestions inline, but I don't think any of them are blocking. Thanks for the fix!

File: packages/perseus-core/src/parse-perseus-json/general-purpose-parsers/defaulted.ts [hunk 5]
```diff
@@ -13,3 +13,33 @@ export function defaulted<T>(
...
+            // If empty or whitespace-only, use fallback
+            if (rawValue.trim() === "") {
+                return ctx.success(fallback());
+            }
+            // Valid non-empty string, use as-is
+            return ctx.success(rawValue);
+        }
+
+        // Invalid type, expect string
+        return ctx.failure("string", rawValue);
```
Comment (benchristel) [inline packages/perseus-core/src/parse-perseus-json/general-purpose-parsers/defaulted.ts:43]: I don't think this "invalid type" case is covered by a test. Does anything fail if you replace this line with `return ctx.success("")`?
